### PR TITLE
70_HaoZhou_Assignment3

### PR DESCRIPTION
**Pull request by mblsktxy**
@zyzlouis

### CODE REVIEW

File: Lesson3/assignment/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+- Call updateEmployee("0xca35b7d915458ef540ade6068dfe2f44e8fa733c",2):  
+	![](https://github.com/mblsktxy/guigulive-operation/blob/master/Lesson3/assignment/images/updateEmployee.png)
+
+#### Problem 2
+
+Please refer to the code
+
+#### Problem 3
+
+```
```
Comment (zyzlouis) [inline Lesson3/assignment/README.md]: C3 linearization中，有一个继承顺序你理解反了：
比如 contract K1 is A, B {...} 的意思是K1继承自B和A，同时B是离K1最近的继承关系(most-derived), 所以：

L(O) = [O]
L(A) = [A, O]
L(B) = [B, O]
L(C) = [C, O]
L(K2) = [K2] + merge(L(C), L(A), [C, A])
= [K2] + merge([C, O], [A, O], [C, A])
= [K2, C] + merge([O], [A, O], [A])
= [K2, C, A, O]
L(K1) = [K1] + merge(L(B), L(A), [B, A])
= [K1] + merge([B, O], [A, O], [B, A])
= [K1, B] + merge([O], [A, O], [A])
= [K1, B, A, O]

L(Z) = [Z] + merge(L(K2), L(K1), [K2, K1])
= [Z] + merge([K2, C, A, O], [K1, B, A, O], [K2, K1])
= [Z, K2] + merge([C, A, O], [K1, B, A, O], [K1])
= [Z, K2, C] + merge([A, O], [K1, B, A, O], [K1])
= [Z, K2, C, K1] + merge([A, O], [B, A, O])
= [Z, K2, C, K1, B] + merge([A, O], [A, O])
= [Z, K2, C, K1, B, A, O]

File: Lesson3/assignment/payroll.sol [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        var employee = employees[employeeId];
+
+        _partialPaid(employee);
+        totalSalary = totalSalary.sub(employees[employeeId].salary);
+        employees[employeeId].salary = salary.mul(1 ether);
+        employees[employeeId].lastPayday = now;
+        totalSalary = totalSalary.add(employees[employeeId].salary);
+    }
+
+    function changePaymentAddress(address employeeId, address newEmployeeId) onlyOwner employeeExist(employeeId) {
```
Comment (zyzlouis) [inline Lesson3/assignment/payroll.sol]: changePaymentAddres不应该是onlyOwner，而是应该让员工自己来操作

**Comment by mblsktxy** [comment]
@zyzlouis 改了，谢谢助教

File: Lesson3/assignment/payroll.sol [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    function updateEmployee(address employeeId, uint salary) onlyOwner employeeExist(employeeId) {
+        var employee = employees[employeeId];
+
+        _partialPaid(employee);
+        totalSalary = totalSalary.sub(employees[employeeId].salary);
+        employees[employeeId].salary = salary.mul(1 ether);
+        employees[employeeId].lastPayday = now;
+        totalSalary = totalSalary.add(employees[employeeId].salary);
+    }
+
```
Comment (zyzlouis) [inline Lesson3/assignment/payroll.sol]: 完成的不错，通过了

**Pull request merged**